pkg/oc/admin/prune/images.go

```diff
@@ -643,3 +691,75 @@ func getRegistryClient(clientConfig *restclient.Config, registryCABundle string,
 		Transport: wrappedTransport,
 	}, nil
 }
+
+func getClientAndMasterVersions(client discovery.DiscoveryInterface) (clientVersion, masterVersion *version.Info, err error) {
+	done := make(chan error)
+
+	go func() {
+		defer close(done)
+
+		ocVersionBody, err := client.RESTClient().Get().AbsPath("/version/openshift").Do().Raw()
+		switch {
+		case err == nil:
+			var ocServerInfo version.Info
+			err = json.Unmarshal(ocVersionBody, &ocServerInfo)
+			if err != nil && len(ocVersionBody) > 0 {
+				done <- err
+				return
+			}
+			masterVersion = &ocServerInfo
+
+		case kerrors.IsNotFound(err) || kerrors.IsUnauthorized(err) || kerrors.IsForbidden(err):
+		default:
+			done <- err
+			return
+		}
+	}()
+
+	select {
+	case err, closed := <-done:
+		if strings.HasSuffix(fmt.Sprintf("%v", err), "connection refused") || clientcmd.IsConfigurationMissing(err) || kclientcmd.IsConfigurationInvalid(err) {
+			return nil, nil, err
+		}
+		if closed && err != nil {
+			return nil, nil, err
+		}
+	case <-time.After(time.Second):
```

Comment:
Why do you need this timeout?

Reply:
The code is inspiried by https://github.com/openshift/origin/blob/50389c070c02b01c9315b3779b278e0b9169e204/pkg/oc/cli/cmd/version.go#L95. The `oc version` asks the master for its version with some default timeout.

This call will be much faster than most of the other calls asking for object lists. On the other hand, I don't want to further block error reporting once we know we have failed and master is busy.